paddle/fluid/ir/dialect/pd_manual_op.cc

```diff
@@ -0,0 +1,155 @@
+#include "paddle/phi/infermeta/ternary.h"
+#include "paddle/phi/infermeta/unary.h"
+
+namespace paddle {
+namespace dialect {
+
+OpInfoTuple AddNOp::GetOpInfo() {
+  std::vector<paddle::dialect::OpInputInfo> inputs = {
+      OpInputInfo("inputs",
+                  "ir::VectorType<paddle::dialect::DenseTensorType>",
+                  false,
+                  false,
+                  false)};
+  std::vector<paddle::dialect::OpAttributeInfo> attributes = {};
+  std::vector<paddle::dialect::OpOutputInfo> outputs = {
+      OpOutputInfo("out", "paddle::dialect::DenseTensorType", false, false)};
+  paddle::dialect::OpRunTimeInfo run_time_info =
+      OpRunTimeInfo("", {""}, {""}, {""}, {""}, {}, {});
+
+  return std::make_tuple(inputs, attributes, outputs, run_time_info);
+}
+
+void AddNOp::Verify() {
+  VLOG(4) << "Start Verifying inputs, outputs and attributes for: AddNOp.";
+  VLOG(4) << "Verifying inputs:";
+  {
+    auto input_size = num_operands();
+    PADDLE_ENFORCE_EQ(
+        input_size,
+        1u,
+        phi::errors::PreconditionNotMet(
+            "The size %d of inputs must be equal to 1.", input_size));
+    if (auto vec_type = (*this)->operand(0).type().dyn_cast<ir::VectorType>()) {
+      for (size_t i = 0; i < vec_type.size(); ++i) {
+        PADDLE_ENFORCE(vec_type[i].isa<paddle::dialect::DenseTensorType>(),
+                       phi::errors::PreconditionNotMet(
+                           "Type validation failed for the 0th input."));
+      }
+    } else {
+      PADDLE_ENFORCE(
+          (*this)->operand(0).type().isa<paddle::dialect::DenseTensorType>(),
+          phi::errors::PreconditionNotMet(
+              "Type validation failed for the 0th input."));
+    }
+  }
+  VLOG(4) << "Verifying attributes:";
+  {
+    // Attributes num is 0, not need to check attributes type.
+  }
+  VLOG(4) << "Verifying outputs:";
+  {
+    auto output_size = num_results();
+    PADDLE_ENFORCE_EQ(
+        output_size,
+        1u,
+        phi::errors::PreconditionNotMet(
+            "The size %d of outputs must be equal to 1.", output_size));
+    PADDLE_ENFORCE(
+        (*this)->result(0).type().isa<paddle::dialect::DenseTensorType>(),
+        phi::errors::PreconditionNotMet(
+            "Type validation failed for the 0th output."));
+  }
+  VLOG(4) << "End Verifying for: AddNOp.";
+}
+
+void AddNOp::Build(ir::Builder &builder,             // NOLINT
+                   ir::OperationArgument &argument,  // NOLINT
+                   ir::OpResult inputs) {
+  VLOG(4) << "Builder construction inputs";
+  std::vector<ir::OpResult> argument_inputs = {inputs};
+  argument.AddOperands(argument_inputs.begin(), argument_inputs.end());
+
+  VLOG(4) << "Builder construction attributes";
+
+  VLOG(4) << "Builder construction outputs";
+  ir::VectorType x = inputs.type().dyn_cast<ir::VectorType>();
+  (void)x;
+
+  std::vector<phi::DenseTensor> vec_dense_x;
+  for (size_t i = 0; i < static_cast<size_t>(x.size()); i++) {
```

Comment:
```suggestion
  for (size_t i = 0; i < x.size(); i++) {
```
VectorType.size()返回的就是size_t

Reply:
done